crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -96,14 +98,27 @@ impl Solution {
                             buy: solution.prices
                                 [&fulfillment.order().buy.token.wrap(solution.weth)],
                         };
-                        let fulfillment = fulfillment.with_protocol_fees(prices)?;
-                        trades.push(Trade::Fulfillment(fulfillment))
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade)
                     }
                     order::Kind::Liquidity => {
                         trades.push(trade);
                     }
                 },
-                Trade::Jit(_) => trades.push(trade),
+                Trade::Jit(jit) => {
+                    if cow_amm_addresses.contains(&jit.order().receiver.into()) {
+                        let prices = ClearingPrices {
+                            sell: solution.prices[&jit.order().sell.token.wrap(solution.weth)],
+                            buy: solution.prices[&jit.order().buy.token.wrap(solution.weth)],
+                        };
+                        let trade = trade.with_protocol_fees(prices)?;
+                        trades.push(trade);
+                    } else {
+                        let mut jit = jit.clone();
+                        jit.remove_protocol_fees();
```

Comment:
I took the decision of directly removing the protocol fees for the corner case in which a non CoW AMM address places an JIT order with protocol fees (which should never happen but better to be safe). The reason is because if I don't do this, we would need to check again against the CoW AMM addresses when calculating the `score`, making it more complex. Another option would be to reject any JIT order with protocol fees which aren't coming from CoW AMM addresses. What do you guys think?